site-scrapers/LynnTech.js

```diff
@@ -0,0 +1,154 @@
+
+    // Setup the return object.
+    const results = { availability: {}, hasAvailability: false };
+
+    // Setup the post data to answer the survey questions
+    let startDate = new Date();
+    startDate.setDate(startDate.getDate());
+    // The data can only be returned for one week. Check the calendar for future weeks.
+    for (const i of Array(10).keys()) {
+        // Format the date as 'YYYY-MM-dd'
+        let startDateFormatted = `${startDate.getFullYear()}-${(
+            '0' +
+            (startDate.getMonth() + 1)
+        ).slice(-2)}-${('0' + startDate.getDate()).slice(-2)}`;
+
+        const postData = `id=${siteId}&vt=${vt}&dept=${dept}&view=plain&start=${startDateFormatted}&filters=%7B%22Providers%22%3A%7B%2213300632%22%3Atrue%7D%2C%22Departments%22%3A%7B%22133001025%22%3Atrue%7D%2C%22DaysOfWeek%22%3A%7B%220%22%3Atrue%2C%221%22%3Atrue%2C%222%22%3Atrue%2C%223%22%3Atrue%2C%224%22%3Atrue%2C%225%22%3Atrue%2C%226%22%3Atrue%7D%2C%22TimesOfDay%22%3A%22both%22%7D`;
+        const options = {
+            hostname: 'mychartos.ochin.org',
+            port: 443,
+            path:
+                '/mychart/OpenScheduling/OpenScheduling/GetOpeningsForProvider',
+            method: 'POST',
+            headers: {
+                'Content-Type':
+                    'application/x-www-form-urlencoded; charset=UTF-8',
+                'Content-Length': postData.length,
+                __RequestVerificationToken: verificationToken,
+                Cookie: cookie,
+            },
+        };
+
+        // now request the schedule
+        const schedulePromise = new Promise(resolve => {
+            const req = https.request(options, res => {
+                let body = '';
+                responseHeaders = res.headers;
+                res.on('data', chunk => {
+                    body += chunk;
+                });
+                res.on('end', () => {
+                    if (res.statusCode >= 200 && res.statusCode < 300) {
+                        resolve(JSON.parse(body));
+                    } else {
+                        console.error(
+                            'Error status code returned from ' + res.url
+                        );
+                        resolve();
+                    }
+                });
+            });
+            req.write(postData);
+            req.on('error', e => {
+                console.error('Error making scheduling request : ' + e);
+            });
+            req.end();
+        });
+
+        const postResponse = await schedulePromise;
+        if (postResponse) {
+            allDays = postResponse.AllDays;
+
+            // The AllDays property is an object, not an array.
+            // The property of the object is generated from the date, which we don't know
+            // We need to iterate over all the properties to determine the dates and the slots.
+            for (const dayEntryProperty in allDays) {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
for future reference

Reply:
Thanks for the pointers. I have plenty of experience with java and python. Not as much with javascript, so I'm happy to get the feedback. In this case, the property names of "AllDays" are generated. We don't know (or care) what the properties names are; we just want the objects. The data structure for the `AllDays` property really should be an array rather than an object, but we obviously don't have control over that.